astroid/nodes/scoped_nodes.py

```diff
@@ -2331,13 +2335,26 @@ def ancestors(self, recurs=True, context=None):
         :returns: The base classes
         :rtype: iterable(NodeNG)
         """
+        if recurs and context in self.all_ancestors:
+            yield from self.all_ancestors[context].keys()
+        elif not recurs and context in self.direct_ancestors:
+            yield from self.direct_ancestors[context].keys()
+
         # FIXME: should be possible to choose the resolution order
         # FIXME: inference make infinite loops possible here
-        yielded = {self}
+        yielded = OrderedDict()
```

Comment:
I think this is necessary for Python 3.6 to preserve the order of yielded ancestors: a plain `dict` achieves the same thing from Python 3.7+